File: server/src/validators.js
// Nombre de usuario de la cuenta web (distinto del nick de Minecraft).
const USERNAME_PATTERN = /^[A-Za-z0-9_]{3,20}$/

// Java: 3-16 caracteres alfanuméricos y guion bajo. Bedrock (via Floodgate)
// permite además espacios, puntos y guiones en el gamertag.
const NICK_PATTERN = /^[A-Za-z0-9_ .-]{3,16}$/

// No es una validación exhaustiva de RFC 5322 a propósito: solo filtra
// errores de tipeo obvios. La verificación real es que el correo reciba el enlace.
const EMAIL_PATTERN = /^[^\s@]+@[^\s@]+\.[^\s@]+$/

function isValidUsername(value) {
  return typeof value === "string" && USERNAME_PATTERN.test(value)
}

function isValidNick(value) {
  return typeof value === "string" && NICK_PATTERN.test(value)
}

function isValidPassword(value) {
  return typeof value === "string" && value.length >= 8 && value.length <= 200
}

function isValidEmail(value) {
  return typeof value === "string" && value.length <= 200 && EMAIL_PATTERN.test(value)
}

module.exports = { isValidUsername, isValidNick, isValidPassword, isValidEmail }

File: django_backend/accounts/validators.py
import re

# Mismas reglas que server/src/validators.js en el backend Node original.
USERNAME_PATTERN = re.compile(r"^[A-Za-z0-9_]{3,20}$")
NICK_PATTERN = re.compile(r"^[A-Za-z0-9_ .\-]{3,16}$")  # Java: 3-16 alfanumérico+_; Bedrock/Floodgate también permite espacio, punto, guion
EMAIL_PATTERN = re.compile(r"^[^\s@]+@[^\s@]+\.[^\s@]+$")  # deliberadamente no exhaustivo (RFC 5322 completo)


def is_valid_username(value):
    return isinstance(value, str) and bool(USERNAME_PATTERN.match(value))


def is_valid_nick(value):
    return isinstance(value, str) and bool(NICK_PATTERN.match(value))


def is_valid_password(value):
    return isinstance(value, str) and 8 <= len(value) <= 200


def is_valid_email(value):
    return isinstance(value, str) and len(value) <= 200 and bool(EMAIL_PATTERN.match(value))
